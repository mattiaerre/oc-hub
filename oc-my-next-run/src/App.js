import React, { PropTypes } from 'react';
import moment from 'moment';
import { Wrapper, Title } from './Styled';

/* eslint-disable react/no-array-index-key */
const App = ({ model }) => { // eslint-disable-line arrow-body-style
  return (
    <Wrapper>
      <Title>{`@${model.twitterUsername}'s next run`}</Title>
      <p>{moment(model.when).format('MMMM Do YYYY, h:mm:ss a')}</p>
      <p>
        <a href={model.where.googleMapsShortUrl}>{model.where.googleMapsShortUrl}</a>
      </p>
      <ul>
        {model.note.map((element, index) => (<li key={index}>{element}</li>))}
      </ul>
    </Wrapper>
  );
};

App.propTypes = {
  model: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

export default App;
