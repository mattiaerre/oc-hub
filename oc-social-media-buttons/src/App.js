import React from 'react';
import styled from 'styled-components';

const data = [
  { className: 'icon-twitter', href: 'https://twitter.com/mattiaerre', text: '@mattiaerre' },
  { className: 'icon-instagram', href: 'https://www.instagram.com/mattiaerre2012', text: '@mattiaerre2012' },
  { className: 'icon-github', href: 'https://github.com/mattiaerre', text: 'mattiaerre' },
  { className: 'icon-wordpress', href: 'https://mattiarichetto.wordpress.com/', text: 'mattiarichetto' },
];

const Ul = styled.ul`
  list-style-type: none;
  display: flex;
  flex-wrap: wrap;
`;

const Li = styled.li`
  margin: 0 0.5em;
`;

const App = () => {
  const list = data.map(element => (
    <Li>
      <i className={element.className} />
      <a href={element.href} rel="noopener noreferrer" target="_blank">{element.text}</a>
    </Li>
  ));

  return (
    <Ul>
      {list}
    </Ul>
  );
};

export default App;
