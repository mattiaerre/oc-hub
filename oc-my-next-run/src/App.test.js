import React from 'react';
import { create } from 'react-test-renderer';
import App from './App';

test('App to match snapshot', () => {
  const props = {
    model: {
      twitterUsername: 'mattiaerre',
      when: new Date(2017, 2, 19, 9, 0, 0),
      where: {
        googleMapsShortUrl: 'https://goo.gl/maps/mvsGHqLtiCs'
      },
      note: ['eat', 'run', 'sleep', 'repeat']
    }
  };

  const tree = create(<App {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
