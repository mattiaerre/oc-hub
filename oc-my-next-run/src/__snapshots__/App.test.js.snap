// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App to match snapshot 1`] = `
<section
  className="sc-bwzfXH bOYEsa"
>
  <h1
    className="sc-bdVaJa kbYEP"
  >
    @mattiaerre's next run
  </h1>
  <p>
    March 19th 2017, 9:00:00 am
  </p>
  <p>
    <a
      href="https://goo.gl/maps/mvsGHqLtiCs"
    >
      https://goo.gl/maps/mvsGHqLtiCs
    </a>
  </p>
  <ul>
    <li>
      eat
    </li>
    <li>
      run
    </li>
    <li>
      sleep
    </li>
    <li>
      repeat
    </li>
  </ul>
</section>
`;
